src/utils/device.js

```diff
@@ -96,3 +96,30 @@ module.exports.isLandscape = function () {
 module.exports.isIOSOlderThan10 = function (userAgent) {
   return /(iphone|ipod|ipad).*os.(7|8|9)/i.test(userAgent);
 };
+
+/**
+ * Check if running in a browser.
+ */
+// we need to check a node api that isn't mocked on either side
+// * `require` and `module.exports` are mocked in browser by bundlers
+// * `window` is mocked in node
+var isBrowserEnvironment = !process;
+module.exports.isBrowserEnvironment = isBrowserEnvironment;
+
+/**
+ * Check if running the bundler.
+ */
+// `process` is also mocked by browserify, but has custom properties
+var isBundlerEnvironment = !!(process && process.browser);
+module.exports.isBundlerEnvironment = isBundlerEnvironment;
+
+/**
+ * Check if running in a browser or spoofed browser.
+ */
+var isBrowserLikeEnvironment = isBrowserEnvironment || isBundlerEnvironment;
```

Comment:
Can we narrow it to two?

- `isBrowserEnvironment` - true if the environment is even browser-like.
- `isNodeEnvironment` - Not browser-like.

And then we can export straight up in two lines:

```
module.exports.isBrowserEnvironment = !process || !!(process && process.browser);

module.exports.isNodeEnvironment = !!module.exports.isBrowserEnvironment
```